pkg/runtime/core/trainingruntime_test.go

```diff
@@ -693,6 +693,80 @@ func TestTrainingRuntimeNewObjects(t *testing.T) {
 					Obj(),
 			},
 		},
+		"succeeded to build JobSet with labels and annotations overrides from the TrainJob's PodSpecOverrides.": {
+			trainingRuntime: testingutil.MakeTrainingRuntimeWrapper(metav1.NamespaceDefault, "test-runtime").RuntimeSpec(
+				testingutil.MakeTrainingRuntimeSpecWrapper(testingutil.MakeTrainingRuntimeWrapper(metav1.NamespaceDefault, "test-runtime").Spec).
+					WithMLPolicy(
+						testingutil.MakeMLPolicyWrapper().
+							WithNumNodes(100).
+							Obj(),
+					).
+					InitContainer(constants.DatasetInitializer, "override-init-container", "test:runtime").
+					InitContainer(constants.Node, "override-init-container", "test:runtime").
+					Container(constants.Node, constants.Node, "test:runtime", []string{"runtime"}, []string{"runtime"}, resRequests).
+					Container(constants.Node, "override-container", "test:runtime", []string{"runtime"}, []string{"runtime"}, resRequests).
+					Obj(),
+			).Obj(),
+			trainJob: testingutil.MakeTrainJobWrapper(metav1.NamespaceDefault, "test-job").
+				UID("uid").
+				RuntimeRef(trainer.SchemeGroupVersion.WithKind(trainer.TrainingRuntimeKind), "test-runtime").
+				Trainer(
+					testingutil.MakeTrainJobTrainerWrapper().
+						Container("test:trainjob", []string{"trainjob"}, []string{"trainjob"}, resRequests).
+						Obj(),
+				).
+				PodSpecOverrides([]trainer.PodSpecOverride{
+					{
+						TargetJobs:  []trainer.PodSpecOverrideTargetJob{{Name: constants.DatasetInitializer}},
+						Labels:      map[string]string{"k1": "v1"},
+						Annotations: map[string]string{"a1": "v1"},
+					},
+					{
+						TargetJobs:  []trainer.PodSpecOverrideTargetJob{{Name: constants.Node}},
+						Labels:      map[string]string{"k2": "v2"},
+						Annotations: map[string]string{"a2": "v2"},
+					},
+				}).
+				Obj(),
+			wantObjs: []runtime.Object{
+				func() runtime.Object {
+					js := testingutil.MakeJobSetWrapper(metav1.NamespaceDefault, "test-job").
+						ControllerReference(trainer.SchemeGroupVersion.WithKind(trainer.TrainJobKind), "test-job", "uid").
+						Replicas(1, constants.DatasetInitializer, constants.ModelInitializer, constants.Node).
+						Parallelism(1, constants.DatasetInitializer, constants.ModelInitializer).
+						Completions(1, constants.DatasetInitializer, constants.ModelInitializer).
+						NumNodes(100).
+						InitContainer(constants.DatasetInitializer, "override-init-container", "test:runtime").
+						InitContainer(constants.Node, "override-init-container", "test:runtime").
+						Container(constants.Node, constants.Node, "test:trainjob", []string{"trainjob"}, []string{"trainjob"}, resRequests).
+						Container(constants.Node, "override-container", "test:runtime", []string{"runtime"}, []string{"runtime"}, resRequests).
+						Obj()
+					for i, rj := range js.Spec.ReplicatedJobs {
```

Comment:
Please can you use create function that adds label and annotation to the desired rJob pod template ?
The function should be similar to this one, but set the label to the pod template:
https://github.com/kubeflow/trainer/blob/aebc0d3ea92a5f9348b6215703df50710a56aceb/pkg/util/testing/wrapper.go#L407